internal_worker.go

```diff
@@ -160,6 +161,31 @@ func ensureRequiredParams(params *workerExecutionParameters) {
 	}
 }
 
+// verifyDomainExist does a DescribeDomain operation on the specified domain
+// It returns an error, if the server returns an EntityNotExist or BadRequest error
+// On any other transient error, this method will just return success
+func verifyDomainExist(client m.TChanWorkflowService, domain string, logger *zap.Logger) error {
+
+	ctx, cancel := newTChannelContext()
+	defer cancel()
+
+	_, err := client.DescribeDomain(ctx, &shared.DescribeDomainRequest{Name: &domain})
+	if err != nil {
+		if _, ok := err.(*shared.EntityNotExistsError); ok {
+			logger.Error("domain does not exist", zap.String("domain", domain), zap.Error(err))
+			return err
+		}
+		if _, ok := err.(*shared.BadRequestError); ok {
+			logger.Error("domain does not exist", zap.String("domain", domain), zap.Error(err))
+			return err
+		}
+		// on any other error, just return true
+		logger.Warn("unable to verify if domain exist", zap.String("domain", domain), zap.Error(err))
```

Comment:
newTChannelContext() comes with a defaultRetryOption - which will retry on transient connect errors. As for InternalServiceErrors, yes, I can add a retry, but the expiry will be < 10 secs. We cannot block worker.Start() for a long time.